docs/onboarding/ck.hiring_process.how_to_guide.md

```diff
@@ -0,0 +1,216 @@
+# Hiring process
+
+<!-- toc -->
+
+- [Hiring stages](#hiring-stages)
+- [HiringMeister](#hiringmeister)
+- [Step by step](#step-by-step)
+- [Warm-up tasks](#warm-up-tasks)
+- [Take-home message for interns](#take-home-message-for-interns)
+- [Intern scoring](#intern-scoring)
+
+<!-- tocstop -->
+
+## Hiring stages
+
+- We separate the hiring process into the following stages:
+- **Intern**
+  - Not a part of the company
+  - Paid, but not through the company payroll
+  - Interns only have access to public repos (e.g., `helpers`, `tutorials`)
+  - Can be full-time or part-time
+    - Since most are students, during summer they can work full-time
+    - In any case we expect at least 20 hrs/week
+  - Interns go through
+    ["light" on-boarding](/docs/onboarding/intern.onboarding_checklist.reference.md)
+    - Read the "must-read" docs about how we do things
+    - Set up for development on their laptops
+  - Interns get [simple tasks](#warm-up-tasks) to work on, so that we can assess
+    how well they can follow directions and work independently
+    - The tasks would be most likely related to Linter, documentation, AI, DS
+    - How to test potential hires to the Infra team? We decide case by case
+  - Ideal goal: minimal effort for us to give a chance to somebody, high
+    probability of failing
+  - Interns are managed only by the hiring team without any input from team
+    leaders
+  - After 1 month, we make a decision whether the intern can move on to the
+    following stage of the hiring pipeline
+- **3-month trial**
+  - Before an intern can turn into a permanent team member, they go through a
+    3-month trial period
+  - At this stage, they work full-time and have all the markings of a Causify
+    team member
+  - They go through
+    [full on-boarding](/docs/onboarding/all.onboarding_checklist.reference.md),
+    receiving company e-mail, server access, access to private repos, etc.
+  - After 3 months, we make a decision whether to make an offer or not
+    - The probability of failing at this stage should be low
+  - This stage is run by the hiring team with some input from team leaders
```

Comment:
- 3-month team members receive feedback on various skills every month to make sure they are adapting and improving in the way we expect